riscv/csr_init.cc

```diff
@@ -430,6 +430,14 @@ void state_t::csr_init(processor_t* const proc, reg_t max_isa)
     auto sireg = std::make_shared<sscsrind_reg_csr_t>(proc, CSR_SIREG, siselect);
     add_ireg_proxy(proc, sireg);
     add_supervisor_csr(CSR_SIREG, std::make_shared<virtualized_indirect_csr_t>(proc, sireg, vsireg));
+    if (proc->extension_enabled_const(EXT_ZICNTR)) {
+      sireg->add_ireg_proxy(SISELECT_SMCDELEG_START, mcycle);
+      sireg->add_ireg_proxy(SISELECT_SMCDELEG_INSTRET, minstret);
+    }
+    if (proc->extension_enabled_const(EXT_ZIHPM)) {
+      for (size_t j = 0; j < (SISELECT_SMCDELEG_END - SISELECT_SMCDELEG_HPMEVENT_3 + 1); j++)
+        sireg->add_ireg_proxy(SISELECT_SMCDELEG_HPMCOUNTER_3 + j, csrmap[CSR_HPMCOUNTER3 + j]);
+    }
```

Comment:
Shouldn't this code have been moved only two lines up, so that it is still conditioned on `if (proc->extension_enabled(EXT_SSCCFG) || proc->extension_enabled(EXT_SMCDELEG))`?

Reply:
Hi, while you are correct that this should be gated by:
```c
    if (proc->extension_enabled(EXT_SSCCFG) || proc->extension_enabled(EXT_SMCDELEG)) {
```
and I've added a check for this, the code cannot be moved just 2 lines up because the scope for `auto sireg` is inside the for loop in that suggesstion, which would be incorrect. There are two `auto sireg` variables with different scopes, one refers to CSR_SIREG, and the other refers to CSR_SIREG2, CSR_SIREG3.... and so on.

Reply:
Makes sense.  Thanks for fixing it.